src/wavm/WAVMWasmModule.cpp

```diff
@@ -1008,198 +1008,56 @@ U32 WAVMWasmModule::mmapFile(U32 fd, size_t length)
     return wasmPtr;
 }
 
-U32 WAVMWasmModule::growMemory(size_t nBytes)
+bool WAVMWasmModule::doGrowMemory(uint32_t pageChange)
```

Comment:
This diff is very hard to read, but I have left the sense-checks that we can only do in WAVM. You can see the new code [here](https://github.com/faasm/faasm/blob/wasm-memory/src/wavm/WAVMWasmModule.cpp#L1011-L1059).